spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkPushedDownAggregateUtil.java

```diff
@@ -0,0 +1,373 @@
+import org.apache.iceberg.expressions.Literal;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.spark.SparkTableUtil;
+import org.apache.iceberg.types.Conversions;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.types.Decimal;
+import scala.collection.JavaConverters;
+
+/** Helper methods for working with Spark aggregate push down. */
+public class SparkPushedDownAggregateUtil {
+
+  private SparkPushedDownAggregateUtil() {}
+
+  public static boolean metricsModeSupportsAggregatePushDown(
+      Table table, List<Expression> aggregates) {
+    MetricsConfig config = MetricsConfig.forTable(table);
+    for (Expression aggregate : aggregates) {
+      String colName = AggregateUtil.getAggregateColumnName(aggregate);
+      if (!colName.equals("*")) {
+        MetricsModes.MetricsMode mode = config.columnMode(colName);
+        if (mode.toString().equals("none")) {
+          return false;
+        } else if (mode.toString().equals("counts")) {
+          if (aggregate.op() == Expression.Operation.MAX
+              || aggregate.op() == Expression.Operation.MIN) {
+            return false;
+          }
+        } else if (mode.toString().contains("truncate")) {
+          if (AggregateUtil.getAggregateType(aggregate).typeId() == Type.TypeID.STRING) {
+            if (aggregate.op() == Expression.Operation.MAX
+                || aggregate.op() == Expression.Operation.MIN) {
+              return false;
+            }
+          }
+        }
+      }
+    }
+
+    return true;
+  }
+
+  public static InternalRow[] constructInternalRowForPushedDownAggregate(
+      SparkSession spark, Table table, List<Expression> aggregates, List<Integer> indexInTable) {
+    List<Object> valuesInSparkInternalRow = Lists.newArrayList();
+    Row[] row = SparkPushedDownAggregateUtil.getStatisticRow(spark, table);
+    for (int index = 0; index < aggregates.size(); index++) {
+      Expression aggregate = aggregates.get(index);
+      Type type = AggregateUtil.getAggregateType(aggregate);
+      valuesInSparkInternalRow.add(
+          SparkPushedDownAggregateUtil.getAggregateValue(
+              aggregate, row, indexInTable.get(index), type));
+    }
+
+    InternalRow[] rows = new InternalRow[1];
+    rows[0] = InternalRow.fromSeq(JavaConverters.asScalaBuffer(valuesInSparkInternalRow).toSeq());
+    return rows;
+  }
+
+  public static Row[] getStatisticRow(SparkSession spark, Table table) {
+    Dataset<Row> metadataRows =
+        SparkTableUtil.loadMetadataTable(spark, table, MetadataTableType.DATA_FILES);
+    Dataset dataset =
+        metadataRows.selectExpr(
+            "lower_bounds", "upper_bounds", "record_count", "null_value_counts");
```

Comment:
I have thought about using something like 
```
metadataRows.select(functions.max(functions.col("upper_bounds.1")))
```
but the values in the statistics such as `upper_bounds` are in byte array. I don't think I can get the correct max/min values. 
For example, if I have 3.333 and 5.555, the corresponding values in statistics are [-33, 79, 83, 64] and [-113, -62, -79, 64], if I get max, I will get [-33, 79, 83, 64] which is 3.333. That's why i  select `lower_bounds`, `upper_bounds` and get the max/min by myself.